pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -217,58 +217,96 @@ func ingest(ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.
 // Resume is part of the jobs.Resumer interface.
 func (s *streamIngestionResumer) Resume(resumeCtx context.Context, execCtx interface{}) error {
 	jobExecCtx := execCtx.(sql.JobExecContext)
-	// Start ingesting KVs from the replication stream.
-	// TODO(casper): retry stream ingestion with exponential
-	// backoff and finally pause on error.
-	err := ingest(resumeCtx, jobExecCtx, s.job)
-	if err != nil {
+
+	// The ingestion job should never fail, only pause, as progress should never be lost
+	handleResumeError := func(err error) error {
```

Comment:
💭 Perhaps we can move this out to a top level method. Then maybeRevertToCutoverTimestamp is right at the top of the function, and we can to the comment above calling it that any code between the start of resume and the cutover needs to be looked at very carefully.